File: Inheritance(C++)/hierarchical.cpp
#include<iostream>
class person
	{
	private:
	char name[20];
	long int phno;
	public:
 	void read()
 	{
 		name = "Akash";
		phno = 9283749093;
 	}
 	void show()
 	{
 		cout << "\nName = " << name;
 		cout << "\nPhone = " << phno;
 	}
		};
class student : public person
	{
	private:
 	int rollno;
 	char course[20];
	public:
 	void read()
 	{
 		person::read();
		rollno = 007;
		course = "Computer Science";
 	}
 	void show()
 	{
 		person::show();
 		cout << "\nRoll No. = " << rollno;
 		cout << "\nCourse = " << course;
 	}
		};
class teacher:public person
{
	private:
	char dept_name[10];
	char qual[10];
	public:
	void read()
	{
		person::read();
		dept_name = "CSE";
		qual = "PhD";
	}
	void show()
	{
		person::show();
		cout << "\nDepartement = " << dept_name;
		cout << "\nQualififcation = " << qual;
	}
		};

void main()
	{
	student s1;
	s1.read();
	cout << "\n************************ Displaying Student Information*************";
	s1.show();
	teacher t1;
	t1.read();
	cout << "\n*****************************Displaying Teacher Information************";
	t1.show();

	getch();
	}

/* OUTPUT
************************ Displaying Student Information*************
Name = Akash
Phone = 9283749093
Roll No. = 007
Course = Computer Science
*****************************Displaying Teacher Information************
Name = Akash
Phone = 9283749093
Department = CSE
Qualification = PhD
*/

File: Inheritance(Java)/Hierarchical.java
class Person
{
    private String name;
    private long phno;

    public void read()
    {
        name = "Akash";
        phno = 928374993;
    }

    public void show()
    {
        System.out.println("Name = " + name);
        System.out.println("Phone = " + phno);
    }
}

class Student extends Person
{
    private int rollno;
    private String course;

    public void read()
    {
        super.read();
        rollno = 007;
        course = "Computer Science";
    }

    public void show()
    {
        super.show();
        System.out.println("Roll No. = " + rollno);
        System.out.println("Course = " + course);
    }
}

class Teacher extends Person
{

    private String dept_name;
    private String qual;

    public void read()
    {
        super.read();
        dept_name = "CSE";
        qual = "PhD";
    }

    public void show()
    {
        super.show();
        System.out.println("Departement = " + dept_name);
        System.out.println("Qualififcation = " + qual);
    }
}

class Hierarchical
{

    public static void main(String args[])
    {
        Student s1 = new Student();

        s1.read();
        System.out.println("******************** Displaying Student Information ********************");

        s1.show();

        Teacher t1 = new Teacher();

        t1.read();
        System.out.println("******************** Displaying Teacher Information ********************");

        t1.show();
    }
}


/* OUTPUT

******************** Displaying Student Information ********************
Name = Akash
Phone = 928374993
Roll No. = 7
Course = Computer Science
******************** Displaying Teacher Information ********************
Name = Akash
Phone = 928374993
Departement = CSE
Qualififcation = PhD


*/
